app/bundles/PointBundle/Controller/Api/PointGroupsApiController.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Mautic\PointBundle\Controller\Api;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Mautic\ApiBundle\Controller\CommonApiController;
+use Mautic\ApiBundle\Helper\EntityResultHelper;
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Mautic\CoreBundle\Factory\ModelFactory;
+use Mautic\CoreBundle\Helper\AppVersion;
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Helper\InputHelper;
+use Mautic\CoreBundle\Helper\IpLookupHelper;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Translation\Translator;
+use Mautic\LeadBundle\Model\LeadModel;
+use Mautic\PointBundle\Entity\Group;
+use Mautic\PointBundle\Model\PointGroupModel;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @extends CommonApiController<Group>
+ */
+class PointGroupsApiController extends CommonApiController
+{
+    /**
+     * @var PointGroupModel
+     */
+    protected $model;
```

Comment:
```suggestion
    private PointGroupModel $model;
```
Or is there some weird inheritance thing going on?

Reply:
Yes, we can't define the type here.